x-pack/legacy/plugins/maps/public/layers/layer_wizard_registry.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+
+type LayerWizard = {
+  description: string;
+  icon: string;
+  id: string;
+  isIndexingSource?: boolean;
+  renderWizard({
+    onPreviewSource,
+    inspectorAdapters,
+  }: {
+    onPreviewSource: () => void;
+    inspectorAdapters: unknown;
+  }): unknown;
+  title: string;
+  sourceType: string;
+};
+
+const registry: LayerWizard[] = [];
+
+export function registerLayerWizard(layerWizard: LayerWizard) {
+  registry.push(layerWizard);
```

Comment:
I would have the layer-wizard generate an `id`, iso of putting this requirement on the client.